dialog: Update accessibility notes (#18841)
mdn/content: files/en-us/web/html/element/dialog/index.md

## Accessibility considerations

To ensure accessibility for users of Safari versions below 15.4, consider using a polyfill such as [a11y-dialog](https://a11y-dialog.netlify.app/) as earlier implementations of <dialog> had [usability issues with some forms of assistive technology](https://www.scottohara.me/blog/2019/03/05/open-dialog.html).

When implementing a dialog, it is important to consider the most appropriate place to set user focus, including the to [autofocus](/en-US/docs/Web/HTML/Global_attributes/autofocus) attribute if appropriate.

Ensure the “Escape” key closes the dialog. If multiple dialogs are open, Escape should only close the last shown dialog.

> The <dialog> element is equivalent to [role="dialog"](/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role) with the modal dialog behaving similar to setting [aria-modal="true"](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal). Ensure your implementation of dialog doesn't break expected default behaviors and follows proper labeling recommendations.

Reviewer: ```suggestion
The `<dialog>` element is exposed by browsers similarly to custom dialogs using the ARIA [role="dialog"](/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role) attribute. `<dialog>` elements invoked by the `showModal()` method will have an implicit [aria-modal="true"](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal), where as `<dialog>` elements invoked by the `show()` method, or rendered by use of the `open` attribute or changing the default `display` of a `<dialog>` will be exposed as `[aria-modal="false"]`.  It is recommended to use the appropriate `showModal()` or `show()` method to render dialogs.
```

## Accessibility considerations

> To ensure accessibility for users of Safari versions below 15.4, consider using a polyfill such as [a11y-dialog](https://a11y-dialog.netlify.app/) as earlier implementations of <dialog> had [usability issues with some forms of assistive technology](https://www.scottohara.me/blog/2019/03/05/open-dialog.html).

Reviewer 2: ```suggestion
To ensure accessibility for users of Safari versions below 15.4, consider using a polyfill such as [a11y-dialog](https://a11y-dialog.netlify.app/) as earlier implementations of `<dialog>` had [usability issues with some forms of assistive technology](https://www.scottohara.me/blog/2019/03/05/open-dialog.html).
```

## Accessibility considerations

To ensure accessibility for users of Safari versions below 15.4, consider using a polyfill such as [a11y-dialog](https://a11y-dialog.netlify.app/) as earlier implementations of <dialog> had [usability issues with some forms of assistive technology](https://www.scottohara.me/blog/2019/03/05/open-dialog.html).

> When implementing a dialog, it is important to consider the most appropriate place to set user focus, including the to [autofocus](/en-US/docs/Web/HTML/Global_attributes/autofocus) attribute if appropriate.

Reviewer 2: ```suggestion
When implementing a dialog, it is important to consider the most appropriate place to set user focus, including using the [autofocus](/en-US/docs/Web/HTML/Global_attributes/autofocus) attribute if appropriate.
```

> Ensure the “Escape” key closes the dialog. If multiple dialogs are open, Escape should only close the last shown dialog.

Reviewer 2: ```suggestion
Ensure the <kbd>Escape</kbd> key closes the dialog. If multiple dialogs are open, escape should only close the last shown dialog.
```

> The <dialog> element is equivalent to [role="dialog"](/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role) with the modal dialog behaving similar to setting [aria-modal="true"](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal). Ensure your implementation of dialog doesn't break expected default behaviors and follows proper labeling recommendations.

Reviewer 2: ```suggestion
The `<dialog>` element is equivalent to [role="dialog"](/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role) with modal dialogs behaving similar to having [aria-modal="true"](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal) set. 

Ensure your dialog implementation doesn't break expected default behaviors and follows proper labeling recommendations.
```
